### PR TITLE
fix(context): allow `inject` to be explicitly invoked for class ctor args

### PR DESCRIPTION
**Pull request by bajtos**
See the discussion in https://github.com/strongloop/loopback-next/pull/3617

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/context/src/inject.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export function inject(
   }
   return function markParameterOrPropertyAsInjected(
     target: Object,
-    member: string,
+    member: string | undefined,
```
Comment (raymondfeng) [inline packages/context/src/inject.ts]: The `ParameterDecorator` type from TypeScript is declared as follows:
```ts
declare type ParameterDecorator = (target: Object, propertyKey: string | symbol, parameterIndex: number) => void;
```

Can we use `''` for constructor parameters?

Comment (raymondfeng) [reply packages/context/src/inject.ts]: Interestingly, at runtime, the `propertyKey` is `undefined` in constructor cases.

Comment (bajtos) [reply packages/context/src/inject.ts]: > Interestingly, at runtime, the `propertyKey` is `undefined` in constructor cases.

That's the weird thing!

**Comment by bajtos** [comment]
> Maybe we should report the consistency to TypeScript.

I was considering that, but was lazy to fill the report 🙈 

Here we go: https://github.com/microsoft/TypeScript/issues/33260

**Pull request merged**